ast/BUILD

```diff
@@ -15,6 +15,7 @@ cc_library(
     hdrs = [
         "ArgParsing.h",
         "ast.h",
+        "DuplicateHashKeyCheck.h",
```

Comment:
I opted to _not_ include it in `ast.h` or `ast/Helpers.h`, because it includes `#include "core/errors/desugar.h"`, which is Desugar specific. Open to suggestions

Reply:
Why is this not in `ast/desugar/` then?

Reply:
There was no existing dependency from `//parser/prism` to `//ast/desugar`. Should I add it?

It felt a little odd for the parser to depend on the desugar, but I guess it's an accurate reflection of the strange relationship there is between them right now (while we're actively developing this)